### ISSUE TITLE
Using !($VAR) for shopt warns double quote

### ISSUE THREAD

**Issue by darulbaby** [koalaman/shellcheck#924]
#### For bugs
- Rule Id SC2086: SC2086
- My shellcheck version (`shellcheck --version` or "online"):  0.3.7
- [ Y] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ Y] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [ Y] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ Y] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
shopt -s extglob
VAR="one|two"
DEST="/tmp/test"
mv !($VAR|.*) "$DEST"

```

#### Here's what shellcheck currently says:
mv !($VAR|.*) "$DEST"
               ^-- SC2086: Double quote to prevent globbing and word splitting.

#### Here's what I wanted or expected to see:
shopt extgloba doesn't work if we do double quote.

**Comment by darulbaby** [comment]
In addition to that, for now I continue with disable SC2086 in my script.

**Comment by koalaman** [comment]
This is intended behavior: shellcheck doesn't know that you actually wanted `VAR` to be interpreted as a glob. It would be buggy for `VAR="my file.txt"` or `VAR="The.Matrix.[1999].mkv"`. 

You can either keep disabling SC2086, or you can rewrite it to not construct globs at runtime.
